v1/summary_proxy.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const mwUtil = require('../lib/mwUtil');
+const URI = require('hyperswitch').URI;
+const P = require('bluebird');
+
+const spec = require('hyperswitch').utils.loadSpec(`${__dirname}/summary.yaml`);
+
+module.exports = (options) => {
+    const isNewStorageEnabled = (() => {
+        const checkEnableRegex = options.new_storage_enabled
+            && mwUtil.constructRegex(options.new_storage_enabled);
```

Comment:
Since the options are static, it would be more efficient to construct the regex once on start-up and then use it in the compares, instead of constructing it every time.

Reply:
it's not doing it every time, it's memoized. Just wanted to use a fancy technic :)